Implement Course Completion (screen & review course)

jatinrise opened the pull request:
Hello,

I have raised the PR so that you can check the error I was encountering in the loop. I am currently unable to view my course as a student. I’ve already explained the issue in detail here:
🔗 [Issue #456 – Comment](https://github.com/dream-aim-deliver/dad-e-class-monorepo/issues/456#issuecomment-3387365720)

Currently, if the course is completed, a modal will pop up displaying the completion card and the feedback form.

I would like to request you to please test this functionality and share your feedback — whether it’s working fine or if you encounter any errors. Once the error is resolved, I will proceed to add the completion card to the layout.

Thank you!

alebg commented on apps/platform/src/lib/infrastructure/client/pages/course-completion.tsx:
```diff
@@ -97,6 +112,39 @@ export default function CourseCompletion({ slug }: CourseCompletionProps) {
 
+    // Handle download certificate
+    const handleDownloadCertificate = () => {
+        if (certificateDataViewModel?.mode === 'default') {
```
@anilrise11 @jatinrise I don't understand the status of this PR. I thought I could check already the PDF certificate. 

You need to: (1) create a course; (2) log in with a different account with only 'student' (and 'visitor') role; (3) add in the database a course_student entry with the course ID and the student ID; (3) tweak a bit @jatinrise 's code to force the modal to show (since the backend is mocked). For me this works fine and I can see the modal:

<img width="1782" height="1371" alt="immagine" src="https://github.com/user-attachments/assets/74c79971-f0ea-40cd-8856-971b6fa38160" />

I was trying to debug why the PDF doesn't generate, and it turns out is it because the `handleDownloadCertificate` function does nothing. Can we please know when this will be implemented? I don't see a reason to think this is blocked, we could already implement here the PDF generator.

alebg reviewed (changes requested):
@jatinrise @anilrise11 please check my inline comment

jatinrise commented:
Hello @alebg,

I have added the download certificate functionality, where I created a simple template. I’ve also incorporated some other changes. Please check it out so that we can make any necessary adjustments before merging this branch.
I have also added course-certificate storybook.

Thank you.
